tanjun/checks.py

```diff
@@ -297,6 +297,37 @@ async def __call__(
         return self._handle_result((permissions & self.permissions) == self.permissions)
 
 
+class HasAnyRoleCheck(_Check):
+    __slots__ = ("required_roles",)
+
+    def __init__(
+        self,
+        roles: collections.Sequence[typing.Union[hikari.SnowflakeishOr[hikari.Role], str]] = [],
+        *,
+        error_message: typing.Optional[str] = "You do not have the required roles to use this command!",
+        halt_execution: bool = True,
+    ) -> None:
+        super().__init__(error_message, halt_execution)
+        self.required_roles = roles
+
+    async def __call__(self, ctx: tanjun_abc.Context, /) -> bool:
+        if not ctx.member:
+            return self._handle_result(False)
+
+        member_roles = ctx.member.get_roles()
```

Comment:
> You had mentioned doing that in the __init__ method, but I think we would need async functionality to lookup possible roles to get their id (if provided as a Role.name/str).

This isn't quite what I mean; what i meant was more checking all of the roles provided in the init to see if they're all IDs and then setting a flag to indicate whether they're all IDs or not. Since if they're all IDs then a fast route which just directly compares against member.role_ids could be added